addons/website_mail_group/static/src/snippets/s_group/options.js

```diff
@@ -30,10 +30,6 @@ options.registry.Group = options.Class.extend({
     },
 
     cleanForSave: function () {
-        // Hide the element by default, this class will be removed
-        // if the current user has access to the group
-        this.$target.addClass('d-none');
-
```

Comment:
@can-odoo After this diff, we now have the `this.$el.removeClass('d-none');` (that you removed earlier) which is there for no apparent reason. In a refactoring, someone will remove it and recreate the disappearing snippet bug. Please, at the very least, it should have been documented to explain why it is there for compatibility. But I think the right solution here is rather to make the same diff as in 14.0: add the d-none class in the `destroy`. Then the `empty()` stuff makes no sense to me: add a comment above it `// TODO this seems useless as the snippet is hidden already when the RPC is made: if there is no response, we just need to not remove the d-none class`. Then in the forward-ported version in master, we'll remove that "empty" code.

Reply:
@qsm-odoo, actually I am trying to understand why with removing `this.$el.removeClass('d-none');` makes snippet disappear.

> But I think the right solution here is rather to make the same diff as in 14.0: add the d-none class in the destroy

I have tried this solution in the first place but with the `destroy`, the snippet is disappeared at the time of `edit`

Reply:
> I have tried this solution in the first place but with the destroy, the snippet is disappeared at the time of edit

Mmh ok, that's because the `MailGroup` widget is not executed in edit mode... but then the existing code is just really buggy as it relies on the fact the d-none class is removed before entering edit mode but not re-added on destroy :grimacing: Check why it worked in 14.0 :shrug: In any case, this can't stay like this: the snippet is not meant to be shown during page load, just removing cleanForSave will save it shown.

> actually I am trying to understand why with removing this.$el.removeClass('d-none'); makes snippet disappear.

Because the snippet is saved hidden (since the cleanForSave add(ed) the d-none class). So if you remove the code that shows the snippet after the RPC during page load... well it stays hidden.

Reply:
@qsm-odoo 
> Check why it worked in 14.0 

For 14.0, after `destroy` gets called while clicking on `edit`, `start` gets called thereafter which removes `d-none` in edit mode.

Whereas for 15.0, destroy gets called and that's it, no snippet available during `edit`, what I have tried is adding `destroy` in `website_mail_group` which will add `d-none`, add destroy for `mail_group` as well which will remove `d-none`.

Reply:
Meh, maybe the least ugly stable solution will then be to not remove the class in destroy for now. But add a comment next the d-none class addition in the start method of the public widget to explain why it is there and why it is not removed in destroy.